consensus/istanbul/backend/test_utils.go

```diff
@@ -144,11 +145,19 @@ func makeBlock(keys []*ecdsa.PrivateKey, chain *core.BlockChain, engine *Backend
 
 	// create the sig and call Commit so that the result is pushed to the channel
 	aggregatedSeal := signBlock(keys, block)
-	err := engine.Commit(block, aggregatedSeal, types.IstanbulEpochValidatorSetSeal{})
+	aggregatedEpochValidatorSetSeal := types.IstanbulEpochValidatorSetSeal{}
+	err := engine.Commit(block, aggregatedSeal, aggregatedEpochValidatorSetSeal)
 	if err != nil {
 		return nil, err
 	}
-	block = <-results
+
+	h := block.Header()
```

Comment:
How come this new code block needs to be added here?

Reply:
Previously `engine.Commit` will search newly added `BlockProcessResult` in the cache to return the block. But since the default setting for test_backend is non-proposer. The block is never saved in the `engine.Seal`.

In my new commit, this has been fixed.